examples/MAML-RL/maml_torchrl.py

```diff
@@ -0,0 +1,236 @@
+
+
+def a2c_loss(traj, policy_module, value_module, value_coef):
+    dist, *_ = policy_module.get_dist(traj)
+    action = traj.get('action')
+    log_probs = dist.log_prob(action)
+
+    traj = traj.exclude('state_value')
+    value_module.module[-1](traj)  # will read the "hidden" key and return a state value
+    value = traj.get('state_value')
+
+    reward = traj.get('reward')
+    done = traj.get('done')
+    next_traj = step_tensordict(traj)
+    next_value = value_module(next_traj).get('state_value').detach()
+
+    # Work backwards to compute `G_{T-1}`, ..., `G_0`.
+    # tderror = TDEstimate(GAMMA, value_module, gradient_mode=True)
+    # tderror = TDLambdaEstimate(GAMMA, LAMBDA, value_module, gradient_mode=True)
+    advantage = vec_td_lambda_advantage_estimate(GAMMA, LAMBDA, value, next_value, reward, done)
+    action_loss = -(advantage.detach() * log_probs.view_as(advantage)).mean()
+    value_error = advantage
+    value_loss = value_error.pow(2).mean()
+
+    assert action_loss.requires_grad
+    assert advantage.requires_grad
+    assert not action.requires_grad
+    assert value_loss.requires_grad
+
+    loss = action_loss + value_coef * value_loss
+    return loss
+
+
+def evaluate(env, dummy_env, seed, task_num, actor_critic, policy, value):
+    pre_reward_ls = []
+    post_reward_ls = []
+    env.reset()
+    device = next(actor_critic.parameters()).device
+
+    inner_opt = torchopt.MetaSGD(actor_critic, lr=0.1)
+
+    tasks = dummy_env.sample_tasks(num_tasks=task_num)
+
+    policy_state_dict = torchopt.extract_state_dict(actor_critic)
+    optim_state_dict = torchopt.extract_state_dict(inner_opt)
+    for idx in range(task_num):
+        env.reset_task(tasks[idx])
+        for _ in range(inner_iters):
+            with set_exploration_mode('random'), torch.no_grad(), timeit('rollout_eval'):
+                pre_traj_td = env.rollout(policy=policy, max_steps=TRAJ_LEN).to(device)
+            inner_loss = a2c_loss(pre_traj_td, policy, value, value_coef=0.5)
+            inner_opt.step(inner_loss)
+        with set_exploration_mode('random'), torch.no_grad(), timeit('rollout_eval'):
+            post_traj_td = env.rollout(policy=policy, max_steps=TRAJ_LEN).to(device)
+
+        # Logging
+        pre_reward_ls.append(torch.sum(pre_traj_td.get('reward'), dim=1).mean().item())
+        post_reward_ls.append(torch.sum(post_traj_td.get('reward'), dim=1).mean().item())
+
+        torchopt.recover_state_dict(actor_critic, policy_state_dict)
+        torchopt.recover_state_dict(inner_opt, optim_state_dict)
+    return pre_reward_ls, post_reward_ls
+
+
+def main(args):
+    device = args.device if torch.cuda.device_count() else args.device
+    # init training
+    torch.manual_seed(args.seed)
+    torch.cuda.manual_seed_all(args.seed)
+    # Env
+    lambda_env = lambda: GymEnv(
+        'TabularMDP-v0',
+        num_states=STATE_DIM,
+        num_actions=ACTION_DIM,
+        max_episode_steps=TRAJ_LEN,
+        seed=args.seed,
```

Comment:
Let's avoid that: all envs will have the same seed! (that's my n1 hypothesis regarding the diff in perf between the 2 implementations)